zigzag/hardware/architecture/core.py

```diff
@@ -121,7 +121,7 @@ def __jsonrepr__(self):
         return json_repr_handler(self.__dict__)
 
     def __hash__(self) -> int:
-        return hash(self.id)
+        return hash_sha512(self.id)
```

Comment:
```suggestion
        return self.id
```
Might as well return the id itself, instead of calling the hash each time